backend/src/bundles/achievements/achievement.entity.ts

```diff
@@ -0,0 +1,112 @@
+import {
+    type ActivityType,
+    type Entity,
+    type Metric,
+    type ValueOf,
+} from './enums/enums.js';
+
+class AchievementEntity implements Entity {
+    private 'id': number | null;
+
+    private 'name': string;
+
+    private 'activity': ValueOf<typeof ActivityType> | null;
```

Comment:
maybe worth to rename it to activityType aswel

Reply:
Thanks for checking. I will rename it.